lib/sales_engine.rb

```diff
@@ -12,6 +14,7 @@ def self.from_csv(data)
   def initialize(data)
     @items = ItemRepository.new(data[:items], self)
     @merchants = MerchantRepository.new(data[:merchants], self)
+    @invoices = InvoiceRepository.new(data[:invoices], self)
```

Comment:
I wonder if initialize and self.from_csv should be switched? what do you think?

Reply:
Do you mean the order? I think self should be first because that's what is being called first. Why do you think they should be switched?